lib/nats.js

```diff
@@ -801,9 +817,12 @@ Client.prototype.flush = function(opt_callback) {
 
 Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (!msg) { msg = EMPTY; }
+  if (typeof subject !== 'string') {
+    this.emit('error', PUB_BAD_SUBJECT_ERR);
+  }
```

Comment:
Indeed; this is not a connection-oriented error.  This is exactly where the callback should be used and a case where a connection error should definitely _not_ be used.